### ISSUE TITLE
`\snippet{doc}` inclusion fails with `#` sections

### ISSUE THREAD

**Issue by mcondarelli** [doxygen/doxygen#9370]
**Describe the bug**
`#` initiated sections are not (correctly?) recognized when included via `\snippet{doc}`.

**Expected behavior**
Included and inline snippets should have a similar look.

**Screenshots**
First screenshot was generated with Doxygen 1.9.1 (Debian Sid standard) and included snipped appears to ignore completely Markdown:
![image](https://user-images.githubusercontent.com/870849/171992382-0254d4a3-2b8b-4071-a1a8-fca1a8df2b32.png)

Second screenshot was generated using self-compiled Doxygen 1.9.5 (git HEAD) and it seems to **delete** included snippet (after first `#`):
![image](https://user-images.githubusercontent.com/870849/171992460-81552685-1e45-4cad-935d-d7d06f87a046.png)

**To Reproduce**
Attached self contained example allows reproduce the problem.
[example.tar.gz](https://github.com/doxygen/doxygen/files/8837341/example.tar.gz)

**Comment by albert-github** [comment]
In the `\snippet{doc}` description theer is a note saying:
> When using the {doc} option, some commands like \cond and \if don't work with this command due to the
moment of parsing.

I think this is also relevant for the sectioning commands  as this is handled also beforehand.
The resulting output of the markdown processing is OK:
```
======== Markdown =========
---- input -------
The text before first hash is actually rendered.

# Some snippet title 1 hash
the title above and this will be preserved in  documentation,
but also here normal text will follow title  without newline.


---- output -----
The text before first hash is actually rendered.

@section autotoc_md1 Some snippet title 1 hash
the title above and this will be preserved in  documentation,
but also here normal text will follow title  without newline.
```

(I just used one level which is enough to show the problem).

**Comment by mcondarelli** [comment]
Sorry, 
I don't fully understand:
If that's the problem I should see the generated command or, at most, have the offending line removed.
Here problem is processing is completely stopped; nothing in the `\snippet{doc}` is processed.

Am I missing something?

Anyway this is minor for me as I can easily use `<h#>...</h#>` constructs which seem to work fine.
I will do some more testing while I write the actual documentation.
Thanks.

**Comment by albert-github** [comment]
The section handling (by means of `@section` type commands) is a bit complex and results in some extra lists. As the new sections are not in the list they are, probably, not seen by doxygen.

Good that you have a workaround.

**Comment by albert-github** [comment]
I reran the example ad with the 1.9.1 version the problem is shown.

With the current version the problem is gone as well though the word `this` is not recognized, with the current master version (1.11.0 (eb110639dcc32804b59abdf0dcb343abde211b1e)) this is already fixed.

Due to the fact that `this` was not yet working in the release setting to fixed but not released.
(please don't close the issue as this will be done at the moment of an official release).

**Comment by mcondarelli** [comment]
Thanks.
I'll retest (but I already worked around the problem).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
